site/docs/samples/notebook.md

```diff
@@ -1,162 +1,219 @@
+      ```
+    1. Open [http://localhost:9000](http://localhost:9000) and login with the printed access key and secret key:
+      ```
+      ACCESS_KEY=$(kubectl get secret minio -o jsonpath="{.data.accesskey}" | base64 --decode)
+      SECRET_KEY=$(kubectl get secret minio -o jsonpath="{.data.secretkey}" | base64 --decode)
+      echo "Access Key: ${ACCESS_KEY}"
+      echo "Secret Key: ${SECRET_KEY}"
+      ```
+    1. Click the :fontawesome-solid-plus-circle: button in the bottom right corner and then **Create bucket** to create a bucket (e.g. "demo").
+    1. Click the :fontawesome-solid-plus-circle: button again and then **Upload files** to upload the CSV file to the newly created bucket.
     
-    Note: The path is a reference to the Egeria metadata server and asset id. In the default Egeria installation `mds1` is the metadata server name.
-
-    - Finally, kill the port-forward
-
-        ```bash
-        kill $!
-        ```
-
-## Reviewing the policies for the dataset
+## Register the dataset in a data catalog
+
+Register the credentials required for accessing the dataset. Replace the values for `access_key` and `secret_key` with the values from the object storage service that you used and run:
+
+```yaml
+cat << EOF | kubectl apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  name: data-csv
+type: Opaque
+stringData:
+  access_key: "${ACCESS_KEY}"
+  secret_key: "${SECRET_KEY}"
+EOF
+```
+
+Then, register the data asset itself in the catalog. Replace the values for `endpoint`, `bucket` and `objectKey` with values from the object storage service that you used and run:
+
+```yaml
+cat << EOF | kubectl apply -f -
+apiVersion: katalog.m4d.ibm.com/v1alpha1
+kind: Asset
+metadata:
+  name: data-csv
+spec:
+  secretRef: 
+    name: data-csv
+  assetDetails:
+    dataFormat: csv
+    connection:
+      type: s3
+      s3:
+        endpoint: "http://minio.m4d-notebook-sample.svc.cluster.local:9000"
+        bucket: "demo"
+        objectKey: "data.csv"
+  assetMetadata:
+    geography: us-south
+    tags:
+    - finance
+    componentsMetadata:
+      nameOrig: 
+        tags:
+        - PII
+      oldbalanceOrg:
+        tags:
+        - sensitive
+      newbalanceOrig:
+        tags:
+        - sensitive
+EOF
+```
+
+The asset is now registered in the catalog. The identifier of the asset is `m4d-notebook-sample/data-csv` (i.e. `<namespace>/<name>`). You will use that name in the `M4DApplication` later.
+
+Notice the `assetMetadata` field above. It specifies the dataset geography and tags. These attributes are later used in policies.
+
+
+## Define data access policies
 
 Currently predefined policies are included as part of the OPA deployment.
 Included are policies that are triggered for datasets that are tagged with 'finance' and have columns `nameOrig` and `nameDest`. The policies indicate that these columns must be redacted (masked) when data is read.
 
 The policies can be found at `third_party/opa/opa-policy.rego`.
```

Comment:
```suggestion
The policies can be found at `third_party/opa/data-and-policies/user-created-policy-2`. To load this policy execute the following steps:

- Open a terminal and clone the [mesh for data repository](https://github.com/IBM/the-mesh-for-data)
- Go to third_party/opa folder in the cloned directory in the terminal 
- Login to the appropriate kubernetes cluster (kind, openshift, etc)
- Shift to the appropriate project in the cluster where opa has been deployed 
- Execute the command  command `make loadpolicy ARGS=data-and-policies/user-created-policy-2`
- Now the above policy will be enforced by opa for any subsequent requests

  
```

Is the above suggestion ok @roee88 ?

Reply:
So user-created-policy-2 match this demo? If so great, I will use your text as inspiration.

Reply:
yes. user-created-policy-2 has user_policies,.rego which has rules that are triggered for datasets that are tagged with 'finance' and have columns `nameOrig` and `nameDest`. The policy indicates that these columns must be redacted (masked) when data is read. Hope it is ok.

Reply:
Perfect. Thanks!